sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusProcessorClientIntegrationTest.java

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.amqp.AmqpRetryOptions;
+import com.azure.core.amqp.AmqpTransportType;
+import com.azure.core.amqp.ProxyOptions;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.implementation.MessagingEntityType;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Duration;
+import java.time.OffsetTime;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.azure.messaging.servicebus.TestUtils.getSessionSubscriptionBaseName;
+import static com.azure.messaging.servicebus.TestUtils.getSubscriptionBaseName;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Tests for {@link ServiceBusProcessorClient}.
+ */
+
+public class ServiceBusProcessorClientIntegrationTest extends IntegrationTestBase {
+    private final ClientLogger logger = new ClientLogger(ServiceBusProcessorClientIntegrationTest.class);
+    private final AtomicInteger messagesPending = new AtomicInteger();
+    private final Scheduler scheduler = Schedulers.parallel();
+
+    private ServiceBusProcessorClient processor;
+    private ServiceBusSenderAsyncClient sender;
+
+    @Override
+    protected void beforeTest() {
+        sessionId = UUID.randomUUID().toString();
+    }
+
+    @Override
+    protected void afterTest() {
+        sharedBuilder = null;
+        try {
+            dispose(processor, sender);
+        } catch (Exception e) {
+            logger.warning("Error occurred when draining queue.", e);
+        }
+    }
+
+    ServiceBusProcessorClientIntegrationTest() {
+        super(new ClientLogger(ServiceBusProcessorClientIntegrationTest.class));
+    }
+
+    /**
+     * Validate that processor receive the message and {@code MaxAutoLockRenewDuration} is set on the
+     * {@link ServiceBusReceiverAsyncClient}. The message lock is released by the client and same message received
+     * again.
+     */
+    @MethodSource("com.azure.messaging.servicebus.IntegrationTestBase#messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveMessage(MessagingEntityType entityType, boolean isSessionEnabled) throws InterruptedException {
+        // Arrange
+        // The message is locked for this duration at a time.
+        final int lockTimeoutDurationSeconds = 15;
```

Comment:
This is set as the tryTimeoutDuration. Why do we need to configure this?

Reply:
'tryTimeoutDuration' is used as 'inactivity time' : means if there is no activity on link (no message arriving in this time), It will close the link. The is defaulted to 60 seconds for my test, so I am setting it to 30 seconds.

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java#L104